### PR TITLE
feat: sentry frontend

### PR DESCRIPTION
**Pull request by barnettx**
## Problem

We want to log errors in the frontend application

## Solution

Add sentry to the frontend

**Features**:

- Error tracking
- Uploading of sourcemaps to sentry

## Deploy Notes

**New environment variables**:

`SENTRY_AUTH_TOKEN`
`FRONTEND_SENTRY_DSN`
`SENTRY_ENVIRONMENT`
`SENTRY_ORG`
`SENTRY_PROJECT`
`SENTRY_URL`

**New dependencies**:

- `@sentry/react`
- `@sentry/tracing`
- `history`: access history of react-router for sentry

**New dev dependencies**:

- `@sentry/webpack-plugin`: uploading of sourcemap to sentry

### CODE REVIEW

**Comment by barnettx** [comment]
Consider implementing Error Boundaries https://docs.sentry.io/platforms/javascript/guides/react/components/errorboundary/

**Comment by barnettx** [comment]
TODO: add credentials once merged

**Review by LoneRifle — APPROVED** [review]
lgtm, merge when ready

**Pull request merged**
